nixos/modules/services/home-automation/homebridge.nix

```
@@ -0,0 +1,549 @@
+      description = ''
+        Configuration options for homebridge config UI plugin.
+
+        For more details, see [the homebridge-config-ui-x documentation](https://github.com/homebridge/homebridge-config-ui-x/wiki/Config-Options).
+      '';
+      type = submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          ## Following parameters must be set, and can't be changed.
+
+          # Must be "config" for UI service to see its config
+          platform = mkOption {
+            type = str;
+            default = "config";
+            description = "Type of the homebridge UI platform";
+            readOnly = true;
+          };
+
+          name = mkOption {
+            type = str;
+            default = "Config";
+            description = "Name of the homebridge UI platform";
+            readOnly = true;
+          };
+
+          # Required to be true to run Homebridge UI as a separate service
+          standalone = mkOption {
+            type = bool;
+            default = true;
+            description = "Whether to run the UI as a standalone service";
+            readOnly = true;
+          };
+
+          # Homebridge can be installed many ways, but we're forcing a double service systemd setup
+          # This command will restart both services
+          restart = mkOption {
+            type = str;
+            default = restartCommand;
+            description = "Command to restart the homebridge UI service";
+            readOnly = true;
+          };
+
+          # Tell Homebridge UI where homebridge is so it can pull package information
+          homebridgePackagePath = mkOption {
+            type = str;
+            default = homebridgePackagePath;
+            defaultText = "/path/to/homebridge";
+            description = "Path to the homebridge package";
+            readOnly = true;
+          };
+
+          # If sudo is true, homebridge will download plugins as root
+          sudo = mkOption {
+            type = bool;
+            default = false;
+            description = "Whether to run the UI with sudo";
+            readOnly = true;
+          };
+
+          # We're using systemd, so make sure logs is setup to pull from systemd
+          log = mkOption {
+            default = { };
+            description = "Log options for homebridge";
+            type = submodule {
+              freeformType = settingsFormat.type;
+              options = {
+                method = mkOption {
+                  type = str;
+                  default = "systemd";
+                  description = "Method to use for logging";
+                  readOnly = true;
+                };
+
+                service = mkOption {
+                  type = str;
+                  default = "homebridge";
+                  description = "Name of the systemd service to log to";
+                  readOnly = true;
+                };
+              };
+            };
+          };
```

I think you don't need to define inner submodules for these options, you should be able to do just e.g.

```suggestion
          log.method = mkOption {
            type = str;
            default = "systemd";
            description = "Method to use for logging";
            readOnly = true;
          };

          log.service = mkOption {
            type = str;
            default = "homebridge";
            description = "Name of the systemd service to log to";
            readOnly = true;
          };
```

---

Changed this here and below in `bridge`

---

Great!